src/mlpack/methods/gmm/eigenvalue_ratio_constraint.hpp

```diff
@@ -89,7 +90,7 @@ class EigenvalueRatioConstraint
     // shouldn't be any negative eigenvalues anyway, so it doesn't matter if we
     // are suddenly forcing them to be positive.  If the first eigenvalue is
     // negative, well, there are going to be some problems later...
-    eigvals = eigvals % ratios;
+    eigvals %= ratios;
```

Comment:
Actually sorry, that should be `eigvals = (eigvals[0] * ratios)` like the diagonal case.  Up to you if you want to fix it here, if not I'll handle it in another PR since `EigenvalueRatioConstraint` is broken anyway.

Reply:
Oh.. sorry. I'll change it right away. Thanks for pointing it out. :)